.github/workflows/full.yml

```diff
@@ -0,0 +1,578 @@
+name: full pipeline
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+  packages: read
+
+env:
+  REGISTRY: ghcr.io
+  CRUX_IMAGE_NAME: dyrector-io/dyrectorio/web/crux
+  CRUX_UI_IMAGE_NAME: dyrector-io/dyrectorio/web/crux-ui
+  DAGENT_IMAGE_NAME: dyrector-io/dyrectorio/agent/dagent
+  CRANE_IMAGE_NAME: dyrector-io/dyrectorio/agent/crane
+  CLI_IMAGE_NAME: dyrector-io/dyrectorio/cli/dyo
+  KRATOS_IMAGE_NAME: dyrector-io/dyrectorio/web/kratos
+  CRUX_WORKING_DIRECTORY: web/crux
+  CRUX_UI_WORKING_DIRECTORY: web/crux-ui
+  GOLANG_WORKING_DIRECTORY: golang
+
+jobs:
+  conventional_commits:
+    name: Conventional commits validation
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup node
+        uses: actions/setup-node@v2
+      - name: Run validation
+        uses: beemojs/conventional-pr-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          config-preset: conventionalcommits
+
+  changes:
+    name: Gather changes and export variables
+    runs-on: ubuntu-22.04
+    needs: conventional_commits
+    outputs:
+      agents: ${{ steps.filter.outputs.agents }}
+      crux: ${{ steps.filter.outputs.crux }}
+      cruxui: ${{ steps.filter.outputs.cruxui }}
+      kratos: ${{ steps.filter.outputs.kratos }}
+      tag: ${{ steps.settag.outputs.tag }}
+      version: ${{ steps.settag.outputs.version }}
+      fallbacktag: ${{ steps.settag.outputs.fallbacktag }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            agents:
+              - 'golang/**'
```

Comment:
You can use your predefined env eg.: `GOLANG_WORKING_DIRECTORY` here, no?

Reply:
Tried, can't. Good catch though, I changed other locations to the variable where I could.

Reply:
Hmmm... Okay, thanks!

Reply:
I was a dumdum, fixed. :))